app/(main)/o/[slug]/conversations/[id]/components/document-summary.tsx

```diff
@@ -0,0 +1,73 @@
+import { Check, Copy } from "lucide-react";
+import { useState } from "react";
+import Markdown from "react-markdown";
+import rehypeHighlight from "rehype-highlight";
+
+interface CodeBlockProps extends React.HTMLAttributes<HTMLPreElement> {
+  children?: React.ReactNode;
+}
+
+interface ReactElement {
+  props: {
+    children?: React.ReactNode;
+  };
+}
+
+const CodeBlock = ({ children, className, ...props }: CodeBlockProps) => {
+  const [copied, setCopied] = useState(false);
+
+  const getCodeContent = (children: React.ReactNode): string => {
+    if (typeof children === "string") return children;
+    if (Array.isArray(children)) {
+      return children.map((child) => getCodeContent(child)).join("");
+    }
+    if (children && typeof children === "object" && "props" in children) {
+      return getCodeContent((children as ReactElement).props.children);
+    }
+    return "";
+  };
+
+  const code = getCodeContent(children).replace(/\n$/, "");
```

Comment:
Make sure to check performance on components like these, particularly how many times it re-renders.

My first reaction was that code should be memoized 
 - const code = useMemo(() => getCodeContent(children).....))

It might be fine if its pretty quick code though.

Here's a good post on it - https://www.joshwcomeau.com/react/why-react-re-renders/